targets/targets.json

```diff
@@ -249,9 +249,9 @@
         "core": "Cortex-M3",
         "supported_toolchains": ["ARM", "uARM", "GCC_ARM", "GCC_CR", "IAR"],
         "extra_labels": ["NXP", "LPC176X", "FLASH_CMSIS_ALGO", "UBLOX_MODEM_GENERIC", "GENERIC_MODEM"],
-        "macros": ["TARGET_LPC1768"],
+        "macros": ["TARGET_LPC1768", "MODEM_ON_BOARD=1"],
```

Comment:
can this be made in the config in the target ? it would have the value, the description and all in one place.

here is a good example: https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L747